oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,293 @@
+      orphaned and cross-repo developer documentation.
+
+    * docs.edx.org_ will be the main landing page for *all* Open edX documentation, including a link to the
+      developer documentation index listed above.
+
+* **Maintainable**. Writers, editors, and reviewers must be able to create and modify documentation without
+  too much effort. Anyone, inside or outside of edX, should have the ability to contribute to documentation.
+
+  * **Co-located**. The closer the documentation lives with its corresponding code, the more likely it will be
+    kept up-to-date.
+
+    * Repo-specific documentation will live within a "docs" folder within its repo. This includes local
+      "decisions", "how-tos", and README.rst. Documentation should not live within the code itself (e.g., in
+      an __init__.py python module).
+
+    * System-wide decisions will live within the open-edx-proposals_ repo.
+
+    * Other system-wide documentation will live within the edx-developer-docs_ repo.
+
+  * **Exported Diagrams**. In order to keep diagrams updated across time, they will need to be exported
+    from whichever tool they were created in, with the exported version maintained in Github.
+
+    * draw.io diagrams will be exported and imported as XML files.
+    * Lucidchart diagrams will be exported and imported as Visio (VDX) files.
+
+* **Version Controlled**. Documentation must be maintained under version control in Github, for the
+  following reasons:
+
+  * **Reviewed**. They will be properly reviewed as defined by that repo's process.
+
+  * **Tagged**. They will be appropriately in-synch with its corresponding code. When an open edX instance
+    is running a specific named release, they can refer to the matching documentation for that version of the
+    codebase.
+
+  * **Source Controlled**. They will have the advantages of having source-control backed files, including sharing
+    across multiple developers and robust storage.
+
+.. _edx-developer-docs: https://github.com/edx/edx-developer-docs
+.. _docs.edx.org: docs.edx.org
+
+Format
+~~~~~~
+
+We will use `Restructured Text (rST)`_ for all developer documentation.
+
+Best practices
+^^^^^^^^^^^^^^
+
+* We strongly recommend keeping line length between 80 and 120 characters. Keeping it less than 120
+  leaves room for future edits.
+
+* Consistency across rST documents is not a requirement.  However, you can use the raw version of this OEP
+  as a reference example.
+
+Learning rST
+^^^^^^^^^^^^
+
+* See the `reStructuredText (rST) Primer`_.
+
+Helpful rST-related tools
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* `Visual Studio Code`_ works pretty well.
+
+* `Online reStructured Text editor`_ helps, but isn't perfect.
+
+* `Online table generator`_ (supports multiple formats).
+
+* `Pandoc`_ can be used to convert between many formats, like Google Doc to rST.
+
+* Other options are listed on stackoverflow_.
```

Comment:
Another option potentially worth mentioning here: PyCharm's rST editor has included a preview panel since 2018.2

Reply:
sound good.  Added a link to https://www.jetbrains.com/help/pycharm/restructured-text.html.